File: selenium-automation/support/selectors/practice_selectors.py
from selenium.webdriver.common.by import By


class PracticeSelectors:
  POST_CONTENT = (By.CSS_SELECTOR, "#main-container .post-content")

  TEST_LOGIN_PAGE_LINK = (
      By.CSS_SELECTOR,
      'a[href*="/practice-test-login/"]',
  )

  TEST_EXCEPTIONS_PAGE_LINK = (
      By.CSS_SELECTOR,
      'a[href*="/practice-test-exceptions/"]',
  )

  TEST_TABLE_PAGE_LINK = (
      By.CSS_SELECTOR,
      'a[href*="/practice-test-table/"]',
  )

File: cypress-automation/cypress/support/features/practice/practice.selectors.js
export const practiceSelectors = {
  content: {
    container: '#main-container',
    article: '#main-container article',
    postContent: '#main-container .post-content',
  },

  links: {
    testLoginPage: 'a[href*="/practice-test-login/"]',
    testExceptionsPage: 'a[href*="/practice-test-exceptions/"]',
    testTablePage: 'a[href*="/practice-test-table/"]',
  },
};

export const practiceLinks = {
  testLoginPage: {
    path: '/practice-test-login/',
    text: 'Test Login Page',
  },
  testExceptionsPage: {
    path: '/practice-test-exceptions/',
    text: 'Test Exceptions',
  },
  testTablePage: {
    path: '/practice-test-table/',
    text: 'Test Table',
  },
};
